### DISCUSSION TITLE
Manage Keycloak migrations

### DISCUSSION THREAD

**Discussion by alirf81** [keycloak/keycloak#14231] [Q&A]
I am planning a Keycloak-based service and need to deliver the Keycloak image with custom configuration to the users.
The problem here is that, as I add more customized configurations to the Keycloak and deliver them to the users, I need to update the old version of deployment to add the new Keycloak configurations.
For example, I might add the custom Authentication flow and need to replace it in the user's Keycloak.
If I use Keycloak's export feature with override enabled, it will remove all users, and groups in the user's Keycloak.
So I was thinking I need to implement the mechanism to manage the state and make the migration scripts between each state(versions), like what they are doing in database migration.
Any ideas or suggestions to implement this would be appreciated.
Thank you!

**Comment by sbellerwork** [comment]
Hello, I currently have the same problem. It would also be interesting for me what would be the best way to perform migrations. 

I could imagine that such a migration as you described is possible using the configuration and the corresponding overrides.

Existing groups and users with password could possibly be copied in the database?

But I think there are much better ways to achieve this.
